update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,750 @@
+	// Degraded means the process of updating the control plane suffers from an observed problem
+	ControlPlaneAssessmentDegraded ControlPlaneAssessment = "Degraded"
+)
+
+// ClusterVersionStatusInsightConditionType are types of conditions that can be reported on ClusterVersion status insight
+type ClusterVersionStatusInsightConditionType string
+
+const (
+	// Updating condition communicates whether the ClusterVersion is updating
+	ClusterVersionStatusInsightUpdating ClusterVersionStatusInsightConditionType = "Updating"
+)
+
+// ClusterVersionStatusInsightUpdatingReason are well-known reasons for the Updating condition on ClusterVersion status insights
+type ClusterVersionStatusInsightUpdatingReason string
+
+const (
+	// CannotDetermineUpdating is used with Updating=Unknown
+	ClusterVersionCannotDetermineUpdating ClusterVersionStatusInsightUpdatingReason = "CannotDetermineUpdating"
+	// ClusterVersionProgressing means that ClusterVersion is considered to be Updating=True because it has a Progressing=True condition
+	ClusterVersionProgressing ClusterVersionStatusInsightUpdatingReason = "ClusterVersionProgressing"
+	// ClusterVersionNotProgressing means that ClusterVersion is considered to be Updating=False because it has a Progressing=False condition
+	ClusterVersionNotProgressing ClusterVersionStatusInsightUpdatingReason = "ClusterVersionNotProgressing"
+)
+
+// VersionMetadataType is the type of the metadata value
+// +kubebuilder:validation:Enum=string;bool;int
+type VersionMetadataType string
+
+const (
+	StringVersionMetadata VersionMetadataType = "string"
+	BoolVersionMetadata   VersionMetadataType = "bool"
+	IntVersionMetadata    VersionMetadataType = "int"
+)
+
+type VersionMetadataKey string
+
+const (
+	// installation denotes a boolean that indicates the update was initiated as an installation
+	InstallationMetadata VersionMetadataKey = "installation"
+	// partial denotes a boolean that indicates the update was initiated in a state where the previous upgrade
+	// (to the original version) was not fully completed
+	PartialMetadata VersionMetadataKey = "partial"
+	// architecture denotes a string that indicates the architecture of the payload image of the version,
+	// when relevant
+	ArchitectureMetadata VersionMetadataKey = "architecture"
+)
+
+type VersionMetadata struct {
+	// +required
+	Key VersionMetadataKey `json:"key"`
+
+	// +required
+	VersionMetadataValue `json:",inline"`
+}
+
+type VersionMetadataValue struct {
+	// +unionDiscriminator
+	// +required
+	// +kubebuilder:validation:Enum:string;bool;int
+	Type VersionMetadataType `json:"type"`
+
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +unionMember
+	String string `json:"string,omitempty"`
+
+	// +optional
+	// +kubebuilder:validation:Type=bool
+	// +unionMember
+	Bool bool `json:"bool,omitempty"`
+
+	// +optional
+	// +kubebuilder:validation:Type=int
+	// +unionMember
+	Integer int `json:"integer,omitempty"`
```

Comment:
you have `InstallationMetadata` and `PartialMetadata` for `bool`, and `ArchitectureMetadata` for `string`.  But I don't see any `VersionMetadataKey` that require an `integer` at the moment.  Can we drop this entry?